File: Quasics2024Code/Crescendo-XrpEnabled/src/main/cpp/commands/SetRobotOdometry.cpp
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

#include "commands/SetRobotOdometry.h"

SetRobotOdometry::SetRobotOdometry(IDrivebase& driveBase, frc::Pose2d pose)
    : m_driveBase(driveBase), m_pose(pose) {
  // Use addRequirements() here to declare subsystem dependencies.
  AddRequirements(&m_driveBase);
}

// Called when the command is initially scheduled.
void SetRobotOdometry::Initialize() {
  m_driveBase.resetOdometry(m_pose);
}

// Returns true when the command should end.
bool SetRobotOdometry::IsFinished() {
  return true;
}

File: Quasics2024Code/CrescendoJava/src/main/java/frc/robot/commands/SetRobotOdometry.java
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

package frc.robot.commands;
import edu.wpi.first.math.geometry.Pose2d;
import edu.wpi.first.wpilibj2.command.Command;
import frc.robot.subsystems.Drivebase;

public class SetRobotOdometry extends Command {
  Drivebase m_drivebase;
  Pose2d m_pose;
  /** Creates a new SetRobotOdometry. */
  public SetRobotOdometry(Drivebase drivebase, Pose2d pose) {
    m_drivebase = drivebase;
    m_pose = pose;
    System.out.println(pose.toString());
    // Use addRequirements() here to declare subsystem dependencies.
    addRequirements(drivebase);
  }

  // Called when the command is initially scheduled.
  @Override
  public void initialize() {
    m_drivebase.resetOdometry(m_pose);
  }

  // Called every time the scheduler runs while the command is scheduled.

  @Override
  public boolean isFinished() {
    return true;
  }
}
